### PR TITLE
TASK: Refactor to sentry version 2

### PR DESCRIPTION
**Pull request by daniellienert**
This refactors the package to use version 2 of the sentry client using the current DSN format.

### CODE REVIEW

File: Classes/Handler/ErrorHandler.php [hunk 1]
```diff
@@ -22,153 +36,117 @@ class ErrorHandler
...
+            foreach ($extraData as $extraDataKey => $extraDataValue) {
+                $scope->setExtra($extraDataKey, $extraDataValue);
+            }
+        });
+
+        \Sentry\captureException($exception);
     }
 
     /**
      * Set tags on the raven context
```
Comment (fnkr) [inline Classes/Handler/ErrorHandler.php]: Nit-pick: I believe they're no longer calling it Raven.

```suggestion
      * Set tags on the Sentry client context
```

File: Classes/Handler/ErrorHandler.php [hunk 2]
```diff
@@ -22,153 +36,117 @@ class ErrorHandler
...
     /**
      * Initialize the raven client and fatal error handler (shutdown function)
      */
     public function initializeObject()
     {
-        $client = new \Raven_Client($this->dsn);
-        $errorHandler = new \Raven_ErrorHandler($client, true);
-        $errorHandler->registerShutdownFunction();
-        $this->client = $client;
+        \Sentry\init(['dsn' => $this->dsn]);
```
Comment (fnkr) [inline Classes/Handler/ErrorHandler.php]:
```suggestion
        if (empty($this->dsn)) {
            return;
        }
        \Sentry\init(['dsn' => $this->dsn]);
```

See comment below.

File: Classes/Handler/ErrorHandler.php [hunk 3]
```diff
@@ -22,153 +36,117 @@ class ErrorHandler
...
      *
      * @param object $exception The exception to capture
      * @param array $extraData Additional data passed to the Sentry sample
      */
-    public function handleException($exception, array $extraData = array())
+    public function handleException($exception, array $extraData = []): void
     {
-        if (!$this->client instanceof \Raven_Client) {
-            return;
-        }
```
Comment (fnkr) [inline Classes/Handler/ErrorHandler.php]: This check was necessary to stop Sentry from throwing "DSN not configured" exceptions in environments where Sentry is not actually used (e.g. in development). Newer versions of the Sentry client [seem to use a `NullTransport`](https://github.com/getsentry/sentry-php/blob/2.1.0/src/ClientBuilder.php#L315-L317) in that case, but I'd still vote to return early.

```suggestion
        if (empty($this->dsn)) {
            return;
        }

```

File: Classes/Handler/ErrorHandler.php [hunk 4]
```diff
@@ -22,153 +36,117 @@ class ErrorHandler
...
+            $scope->setTag('flow_version', FLOW_VERSION_BRANCH);
+        });
     }
 
     /**
-     * Set user information on the raven context
+     * @return string
      */
-    protected function setUser()
+    private function getCurrentUserName(): string
```
Comment (fnkr) [inline Classes/Handler/ErrorHandler.php]: Nit-pick: [Username](https://dictionary.cambridge.org/dictionary/english/username) is an english word and therefore should not be camel-cased. ([One should capitalize the letter after where there would "normally" be a space.](https://softwareengineering.stackexchange.com/a/230159))

```suggestion
    private function getCurrentUsername(): string
```

(Also affects calls to this method and the `$userName` variable below.)

File: Classes/Handler/ErrorHandler.php [hunk 5]
```diff
@@ -22,153 +36,117 @@ class ErrorHandler
...
+        });
+
+        \Sentry\captureException($exception);
     }
 
     /**
      * Set tags on the raven context
      */
-    protected function setTags()
+    private function setTags(): void
```
Comment (fnkr) [inline Classes/Handler/ErrorHandler.php]: Why did you make everything private?

File: Classes/Handler/ErrorHandler.php [hunk 6]
```diff
@@ -22,153 +36,117 @@ class ErrorHandler
...
-
-        $tags = array(
-            'php_version' => phpversion(),
-            'flow_context' => (string)$environment->getContext(),
-            'flow_version' => FLOW_VERSION_BRANCH
-        );
-
-        $this->client->tags_context($tags);
+        Hub::getCurrent()->configureScope(function (Scope $scope): void {
+            $scope->setTag('php_version', 'phpversion()');
```
Comment (fnkr) [inline Classes/Handler/ErrorHandler.php]: I guess this should not be enclosed by quotes?

```suggestion
            $scope->setTag('php_version', phpversion());
```

Comment (daniellienert) [reply Classes/Handler/ErrorHandler.php]: Absolutely!!

File: Classes/Handler/ErrorHandler.php (hunk 5 above)
Comment (daniellienert) [reply Classes/Handler/ErrorHandler.php]: Just because the methods are not meant to be extended

**Pull request merged**